airgun/entities/cloud_inventory.py

```diff
@@ -11,33 +13,59 @@ class CloudInventoryEntity(BaseEntity):
 
     def read(self, entity_name=None, widget_names=None):
         view = self.navigate_to(self, 'All')
+        result = view.read(widget_names=widget_names)
         if entity_name:
-            view.searchbox.search(entity_name)
-            entity_item = next(view.inventory_list[entity_name])
-            return entity_item.read()
-        view.searchbox.clear()
-        return view.read(widget_names=widget_names)
+            view.inventory_list.toggle(entity_name)
+            result.update(view.inventory_list.read())
+        return result
 
     def generate_report(self, entity_name):
         view = self.navigate_to(self, 'All')
-        view.searchbox.search(entity_name)
-        entity_item = next(view.inventory_list[entity_name])
-        entity_item.browser.click(entity_item.generating.restart, ignore_ajax=True)
-        wait_for(lambda: entity_item.status == 'idle', timeout=180, delay=1, logger=view.logger)
+        view.inventory_list.toggle(entity_name)
+        view.browser.click(view.inventory_list.generating.restart, ignore_ajax=True)
+        # FIX:
+        sleep(5)
+        wait_for(
+            lambda: view.inventory_list.status == 'idle', timeout=180, delay=1, logger=view.logger
+        )
 
     def download_report(self, entity_name):
         view = self.navigate_to(self, 'All')
-        view.searchbox.search(entity_name)
-        entity_item = next(view.inventory_list[entity_name])
-        entity_item.browser.click(entity_item.uploading.download_report, ignore_ajax=True)
-        wait_for(lambda: entity_item.status == 'idle', timeout=180, delay=1, logger=view.logger)
+        view.inventory_list.toggle(entity_name)
+        view.browser.click(view.inventory_list.uploading, ignore_ajax=True)
+        view.browser.click(view.inventory_list.uploading.download_report, ignore_ajax=True)
+        # FIX:
+        sleep(5)
```

Comment:
Added the workaround. No longer need to use sleep.

Reply:
Using `view.inventory_list.generating.terminal.read()` instead of `view.inventory_list.status` doesn't seem good. So reverted the fix. Also noticed that the `STATUS` locator was wrong, so fixed it as well.

Reply:
Can we just use the `handle_exception` kwarg for `wait_for` so that the exception doesn't stop the loop?

Reply:
added.